MakeHist/mumu.C

```diff
@@ -0,0 +1,337 @@
+                           lep1_jet2_mass = ((leptons[1])+(Jets[Jet2_nonB])).M();
+                           lep2_jet1_mass = ((leptons[2])+(Jets[Jet1_nonB])).M();
+                           lep2_jet2_mass = ((leptons[2])+(Jets[Jet2_nonB])).M();
+                       }
+
+                       else if(Nbjet == 1){
+                           lep1_jet1_mass = ((leptons[1])+(bJets[0])).M();
+                           lep1_jet2_mass = ((leptons[1])+(Jets[Jet2_nonB])).M();
+                           lep2_jet1_mass = ((leptons[2])+(bJets[0])).M();
+                           lep2_jet2_mass = ((leptons[2])+(Jets[Jet2_nonB])).M();
+                       }
+
+                       else if(Nbjet > 1){
+                           lep1_jet1_mass = ((leptons[1])+(bJets[Jet1_B])).M();
+                           lep1_jet2_mass = ((leptons[1])+(bJets[Jet2_B])).M();
+                           lep2_jet1_mass = ((leptons[2])+(bJets[Jet1_B])).M();
+                           lep2_jet2_mass = ((leptons[2])+(bJets[Jet2_B])).M();
+                       }
+
+                       //SBM calculation
+                       float SBM1 = 0.;
+                       float SBM2 = 0.;
+                       if(abs(lep1_jet1_mass - lep2_jet2_mass) < abs(lep2_jet1_mass - lep1_jet2_mass)){
+                           SBM1 = lep1_jet1_mass;
+                           SBM2 = lep2_jet2_mass;
+                       }
+                       else{
+                           SBM1 = lep2_jet1_mass;
+                           SBM2 = lep1_jet2_mass;
+                       }
+
+                       //Fill max SBM (Top mass bound)
+                       //if(SBM1>170. || SBM2>170.){//>170 cut
+                       if(SBM1<SBM2)SBM_hist->Fill(SBM2,weight);
+                       else SBM_hist->Fill(SBM1,weight);
+
+                       //METvsMmm (Normalized MET)
+                       float Mmm = ((leptons[0])+(leptons[1])).M();
+                       float MET_t = MET[0];
+                       float METvsMmm = MET_t/Mmm ;
+                       //if(METvsMmm<0.2){ // <0.2 cut
+                       METvsMmm_hist->Fill(METvsMmm,weight);
+                       MET_hist->Fill(MET_t);
+                       dilep_mass_hist->Fill(Mmm);
+
+                       //(HT-LT) calculation
+                       float DHTLT = -999.;
+                       if(Nbjet > 1){
+                           DHTLT = (Jet1_B_PT+Jet2_B_PT) - (leptons[0].Pt()+leptons[1].Pt());
+                           //DHTLT = (All_bjet_PT) - (leptons[0].Pt()+leptons[1].Pt());
+                       }
+                       else if(Nbjet == 1){
+                           DHTLT = (bJets[0].Pt()+Jet2_nonB_PT) - (leptons[0].Pt()+leptons[1].Pt());
+                       }
+                       else if(Nbjet == 0){
+                           DHTLT = (Jet1_nonB_PT+Jet2_nonB_PT) - (leptons[0].Pt()+leptons[1].Pt());
+                       }
+
```

Comment:
Same comment as for the lepton jet masses. Once the jet pair is chosen, generalize these to have less if/else statements that are only going to be confusing and clutter up the code.